### ISSUE TITLE
version 1.0.2 alters how zip files are made, which makes them appear to have changed when their contents have not changed.

### ISSUE THREAD

**Issue by j2d3** [terraform-providers/terraform-provider-archive#14]
### Terraform Version
```Terraform v0.11.3
+ provider.archive v1.0.0
+ provider.aws v1.11.0
+ provider.template v1.0.0```
Currently running 1.0.0 provider, before the change introduced by 1.0.2

### Affected Resource(s)
- lambda functions' archive provider generated zip files
- any archive provider generated zip files that have changes tracked by hashes

### Terraform Configuration Files
we have tons, this affects all our account-regions because we have lambda scripts in our base module. This means, anyone with the latest archive-provider will generate a plan that wants to re-upload the lambda zip files, despite no changes in the source files for those zips. This causes a great deal of confusion for other teams, etc. We have temporarily pinned the archive provider version to 1.0.0 to avoid the confusion for now.

### Debug Output

### Panic Output

### Expected Behavior
Since the source files didn't change, the zip files shouldn't have changed either. 

### Actual Behavior
The hashes used to track whether the zip files have changed or not are actually changed when using this provider 1.0.2 vs 1.0.0

### Steps to Reproduce
1. Using a setup that employs the aws archive provider to generate a zip for lambda function that has already been successfully deployed under archive provider 1.0.0, run a plan. There should be no changes. Now, delete the generated zips. Upgrade provider to 1.0.2, and run a plan. Now, tf will want to update all the lambda functions. If you let it, it's not actually an issue. The zips are still valid, and so uploading them again doesn't change anything, as the source files have not changed. However, if one user does this and applies the changes / reuploads the files and stores the new hashes in the state file, then another user with the older provider runs a plan, tf will want to reupload the zips again and change all the hashes back.

Our temporary solution is pin the archiver to 1.0.0. However, I can also imagine forcing it to be 1.0.2 and then running a mass plan / apply (on all 70 or so of our accounts) - updating them all to the "new" hashes. We want to know what has changed about the zip-creating process that alters the files, though. We don't want to do this often!

### Important Factoids
We are running terraform for around 70 different accounts under one repo. This situation happens due to lambdas we have defined in our base-module, so it affects all accounts.

### References

**Comment by pawbur** [comment]
I think the reason is that now file header is included in the archive. File header includes modification time of the file so whenever his time is changes ie: git checkout, header changes and this results in changing checksum of the archive.

**Comment by burdandrei** [comment]
and now if you're running plan with binary output in one place and then run apply in another zip file is not there anymore

**Comment by paultyng** [comment]
I agree this is a bug in the handling, and will push the PR (#15) through. 

As an aside, we do discourage the use of Terraform as a "build" tool to produce your deployment artifacts. We would not recommend someone build a binary, create a VM image, or build a Docker container inside Terraform, as its not a use case that it was designed for and other tools are much better at artifact generation and better satisfy the requirements of such a system.

**Comment by paultyng** [comment]
Fixed in #16
